### PR TITLE
new GSoC idea for license submission tool

### PR DESCRIPTION
**Pull request by jlovejoy**
I need to add more to this, but it's a start

need input from @swinslow and @goneall as well!

### CODE REVIEW

File: 2023_ideas.md [hunk 1]
```diff
@@ -21,6 +21,35 @@ This can be either a short (175 hours) project, implementing only the basic func
...
+* Ability to understand the current process and identify how this will improve efficiency, including make suggestions in addition to what is listed here
+* Planning and ideas for sustainability of the online tools maintenance
+
+### Duration  
+not sure...
+
+### Background Information  
+* https://github.com/spdx/license-list-XML/blob/main/DOCS/request-new-license.md
+* https://github.com/spdx/license-list-XML/issues
+
```
Comment (zvr) [inline 2023_ideas.md:48]: Can we also add a pointer to the source of the current tool? @goneall can you provide this?

File: 2023_ideas.md [hunk 2]
```diff
@@ -21,6 +21,35 @@ This can be either a short (175 hours) project, implementing only the basic func
...
+* automatcally archive or remove the listing from the online tool, once the PR is merged (optional)
+
+
+### Skills Needed  
+* Django and python skills
+* Ability to understand the current process and identify how this will improve efficiency, including make suggestions in addition to what is listed here
+* Planning and ideas for sustainability of the online tools maintenance
+
+### Duration  
+not sure...
```
Comment (zvr) [inline 2023_ideas.md:43]: From the description of the changes to the current tool, it looks to me like a medium-size project (around 175 hours).

**Review by zvr — CHANGES_REQUESTED** [review]

File: 2023_ideas.md [hunk 3]
```diff
@@ -21,6 +21,35 @@ This can be either a short (175 hours) project, implementing only the basic func
...
 
+## SPDX License Submission Online Tool - increase functionality
+
+The SPDX-legal team relies heavily on the "submit new license" online tool accessed at https://tools.spdx.org/app/ which creates a new Issue in the SPDX License List Github repo when someone uses the online tool to submit a new license. 
+
+This online tool also currently has good capability for creating the XML file necessary for licenses that have been accepted to the SPDX License List and submitting a PR to the SPDX License List Github repo. 
+
+It would greatly increase the efficiency of license submissions and ease the workload of the SPDX-legal team if we could add the following fucntionality:
+* generate the .txt test file in the PR, along with the XML file
+* ??
```
Comment (goneall) [inline 2023_ideas.md:32]: Couple of ideas:
- Add support for exceptions
- Add or fix unit tests
- Improve the error handling, especially when Github pull requests fail

Comment (goneall) [reply 2023_ideas.md:32]: What do you think about adding https://github.com/spdx/spdx-online-tools/issues/398?

File: 2023_ideas.md (hunk 2 above)
Comment (goneall) [reply 2023_ideas.md:43]: Agree - medium project

File: 2023_ideas.md (hunk 1 above)
Comment (goneall) [reply 2023_ideas.md:48]: Here's some more links:

- https://github.com/spdx/spdx-online-tools - repo
- https://github.com/spdx/spdx-online-tools/issues/399 - issue to add .txt file
- https://github.com/spdx/spdx-online-tools/issues/398 - exceptions to be added

**Review by goneall — CHANGES_REQUESTED** [review]
Just a few suggestions and additions

**Comment by zvr** [comment]
We can also add a generic text "Various improvements on the tool are documented as GitHub tickets tagged `Submit New License` https://github.com/spdx/spdx-online-tools/issues?q=is%3Aissue+is%3Aopen+label%3A%22Submit+New+License%22"

**Comment by jlovejoy** [comment]
Thanks @goneall and @zvr - I think I've updated to address your suggestions!

**Comment by zvr** [comment]
@jlovejoy can you rebase on the latest version already published? Your changes are still from the older version and therefore present "conflicts" (on the box here), so that merging is blocked.

**Comment by jlovejoy** [comment]
ugh, I don't now how to resolve conflicts...

**Pull request merged**